### DISCUSSION TITLE
Dropping the support for SearX instances on searx.space and in this repository.

### DISCUSSION THREAD

**Discussion by unixfox** [searxng/searx-instances#336] [Announcements]
Hello,

Since SearXNG was forked from SearX 2 years ago, we have always supported SearX instances on https://searx.space as this project is maintained by SearXNG developers: https://github.com/searxng/searx-space. We have made some efforts to keep an active list of both projects: https://github.com/searxng/searx-space/pull/83 and https://github.com/searxng/searx-space/pull/105.

Now fast-forward to today, SearXNG and SearX are two separate projects that don't look like the same project anymore:
- SearXNG actively tries to find and push fixes as soon as possible when an engine is not working anymore, always trying to keep a working user experience. Meanwhile, SearX backports our changes 2 or 3 months later, resulting in a very long broken experience for the user.
- SearXNG user interface doesn't look like the same as SearX anymore.
- The core of SearXNG has vastly changed compared to SearX: HTTPX support for a much more stable experience, metrics for faster reporting of broken engines, built-in limiter for defending against bots and much more.

As searx.space is now filled with almost only SearXNG instances, we have come to the conclusion that we do not want the new users to think that they will get the same user experience on SearXNG and SearX. That's why we will be removing all the SearX instances on the **1st September 2023**.

All the remaining and working SearX instances from the public list will get moved to https://github.com/searx/searx-instances. There will be no update in this repository (searx/searx-instances) from us (even though I still have write access in this repository). We advise adapting your code to this file [searx/searx-instances/searxinstances/instances.yml](https://github.com/searx/searx-instances/blob/master/searxinstances/instances.yml) if you still want to support SearX in your project.

**Comment by 4NUB1S-0N-0P10M** [comment]
i have another suggestion, wouldn't it be better to add an additional category on https://searx.space for SearX instances only?? 
meaning that instead of having both SearX and SearXNG instances cramped together on the same instances list (which also makes it very disorganized), perhaps it's a better idea to separate them both within the same website without getting rid of any SearX links; regardless if they are unnecessary in your perspectives.

**Reply by unixfox** [reply]
It's not only a classification issue.

It's the burner of maintaining the list of instances from a project that we do not manage anymore.

**Comment by return42** [comment]
> i have another suggestion, wouldn't it be better to add an additional category 

We have a lot of additional code to support searx instances on searx.space.  Since Searx instances are more or less broken and we do not get any feedback from the searx team nor support to maintain the code base on searx.space and searx-instances we decided not to recommend searx instances nor maintain any longer the different codes needed for searx.   If you ask me; searx project is dead and searx code is broken, thats why we founded SearXNG.

**Comment by 4NUB1S-0N-0P10M** [comment]
No i wasn't saying anything regarding Searx's status or performance, it's just that i couldn't understand that it would be that much of a burden to also maintain searx insances too. According to your [searx-instances repo](https://github.com/searxng/searx-instances#for-searxng-instance-operators), it states that anyone can: add/remove/edit a searx instance, then get verified by reviewers; how is that a burden if the searx instance owners (or normal users on their behalf) do the main job of contributing to the instance list itself?  
I also do agree that it's partly the Searx team's fault for not providing back any support where it's due and deserved, however i don't think it mean that all the searx instances out there should be dumped in an archived repo (Not exactly sure how many searx instances are out there); at least that's my perspective after all, i don't know how exactly this process would be practically. 
edit: fixed a minor typo

**Reply by return42** [reply]
> it's just that i couldn't understand that it would be that much of a burden to also maintain searx insances too.

We have a lot of code & infrastructure to check uptime and quality of the engines of each instance.  Since searx and SearXNG are different (searx core development has been stopped years ago while SearXNG grows) we have to maintain two different code bases and infrastructures .. and this is a lot of work.

If you navigate on searx.space you will see that it is not just a simple list of URLs, even if it would be such a simple list, we do not want to recommend our users instances that are partial broken or [ have other questionable peculiarities](https://github.com/searxng/searx-instances/discussions/120#discussioncomment-3131060)

> Not exactly sure how many searx instances are out there

- https://github.com/searxng/searx-instances/pull/331/files

round about 13 searx instances on the list, fives haven't been update since the beginning (if the version number they show is correct) .. and we do not have a chance to contact the maintainers because these instances were taken from an old list and there are no registration entries for them here in the issue tracker.

To put it in my words: We want to get rid of the old crap we just inherited and nobody seems to care about.  Whoever wants to be listed on searx.space has to follow our rules and should not make us more work than we already have. Our resources are already very scarce and we want to use them for an active community and not waste them on people who don't care about anything.

**Comment by nikkehtine** [comment]
This might be a crazy suggestion, but... wouldn't it be better to merge the two projects, since they essentially serve the same purpose, yet one has much more functionality? From what I've understood from this post, SearX backports changes from SearXNG anyway, that coupled with most instances running XNG instead of X makes it seem like a good idea to me. Current SearX could be renamed to "Legacy" and XNG would be the main focus.
Of course there is one major issue, which is that both teams will have to reach an agreement on that, and it's going to be quite emotional to see the project you've been working on for so long to be faded out to make place for the successor made by a different team... But the benefit of this would be that there will be less work to maintain the two projects, and yet it would be easier with a bigger team.

**Reply by return42** [reply]
SearXNG is the successor of searX, searX will not be developed further --> there is no function in searX that does not exist in SearXNG (except for the oscar theme, which is no longer used in SearXNG due to its vulnerable libraries). 

To put it a bit flippantly .. the only thing we could still take over from the searX project in the SearXNG project are the malfunctions :-)

**Reply by unixfox** [reply]
SearX and SearXNG are two projects maintained by different group of people, both group do not want to work together.

**Reply by geckolinux** [reply]
@return42 
> there is no function in searX that does not exist in SearXNG

Unfortunately there is:
https://github.com/searxng/searxng/issues/966
That change from the original SearX unfortunately ruined SearXNG for me, as I always want to have instant results from Wolfram Alpha, and I don't run my own instance so as to hack the code to my liking.

**Reply by return42** [reply]
> Unfortunately there is:

@geckolinux: no there isn't :-) .. its not a feature you miss, its a default setup that does not fits well to your daily usage / you can use my instance https://darmarit.org/searx  .. se my  https://github.com/searxng/searxng/issues/966#issuecomment-1652983042

**Reply by geckolinux** [reply]
@return42 

> its not a feature you miss, its a default setup that does not fits well to your daily usage

Hmm, I don't mind if the defaults aren't to my liking. The problem is that as an end user I can't change the settings to make it work as I need (in my case, to have Wolfram Alpha instant answers for every query) due to https://github.com/searxng/searxng/pull/813. The only way currently to make it work as I need is to run my own instance and change the categorization code.

> you can use my instance https://darmarit.org/searx

I am now! :-) Thanks very much!
